The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4442097778/jobs/7797963970) [(plain)](https://github.com/rust-lang/rust/commit/ae66ab2be4254bf23c1171047197fbc5869ed269/checks/12065520663/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking addr2line v0.17.0
error[E0425]: cannot find value `e` in this scope
    --> library/std/src/sys/windows/fs.rs:1241:16
     |
1241 |             if e.raw_os_error() == Some(c::ERROR_CANNOT_ACCESS_FILE as u32) {

error[E0425]: cannot find value `ERROR_CANNOT_ACCESS_FILE` in module `c`
    --> library/std/src/sys/windows/fs.rs:1241:44
     |
     |
1241 |             if e.raw_os_error() == Some(c::ERROR_CANNOT_ACCESS_FILE as u32) {
     |                                            ^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `ERROR_CANT_ACCESS_FILE`
     |
    ::: library/std/src/sys/windows/c/errors.rs:719:1
     |
719  | pub const ERROR_CANT_ACCESS_FILE: DWORD = 1920;
     | ----------------------------------------------- similarly named constant `ERROR_CANT_ACCESS_FILE` defined here
error[E0615]: attempted to take value of method `file_type` on type `FileAttr`
    --> library/std/src/sys/windows/fs.rs:1247:22
     |
     |
1247 |             if attrs.file_type.is_symlink() {
     |
help: use parentheses to call the method
     |
     |
1247 |             if attrs.file_type().is_symlink() {

Some errors have detailed explanations: E0425, E0615.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `std` due to 3 previous errors
```

</details>

--- a/library/std/src/sys/windows/fs.rs
+++ b/library/std/src/sys/windows/fs.rs
@@ -1246,7 +1246,7 @@ pub fn stat(path: &Path) -> io::Result<FileAttr> {
                 }
             }
             Err(err)
-        },
+        }
         Ok(attrs) => Ok(attrs),
     }
 }

--- a/library/std/src/sys/windows/fs.rs
+++ b/library/std/src/sys/windows/fs.rs
@@ -1236,7 +1236,19 @@ pub fn link(_original: &Path, _link: &Path) -> io::Result<()> {
 }
 
 pub fn stat(path: &Path) -> io::Result<FileAttr> {
-    metadata(path, ReparsePoint::Follow)
+    match metadata(path, ReparsePoint::Follow) {
+        Err(err) => {
+            if err.raw_os_error() == Some(c::ERROR_CANT_ACCESS_FILE as i32) {
+                if let Ok(attrs) = lstat(path) {
+                    if !attrs.file_type().is_symlink() {
+                        return Ok(attrs);
+                    }
+                }
+            }
+            Err(err)
+        },
+        Ok(attrs) => Ok(attrs),
+    }
 }
 
 pub fn lstat(path: &Path) -> io::Result<FileAttr> {

--- a/library/std/src/sys/windows/fs.rs
+++ b/library/std/src/sys/windows/fs.rs
@@ -1237,17 +1237,15 @@ pub fn link(_original: &Path, _link: &Path) -> io::Result<()> {
 
 pub fn stat(path: &Path) -> io::Result<FileAttr> {
     match metadata(path, ReparsePoint::Follow) {
-        Err(err) => {
-            if err.raw_os_error() == Some(c::ERROR_CANT_ACCESS_FILE as i32) {
-                if let Ok(attrs) = lstat(path) {
-                    if !attrs.file_type().is_symlink() {
-                        return Ok(attrs);
-                    }
+        Err(err) if err.raw_os_error() == Some(c::ERROR_CANT_ACCESS_FILE as i32) => {
+            if let Ok(attrs) = lstat(path) {
+                if !attrs.file_type().is_symlink() {
+                    return Ok(attrs);
                 }
             }
             Err(err)
         }
-        Ok(attrs) => Ok(attrs),
+        result => result,
     }
 }
 

